src/hkdf.rs

```diff
@@ -39,19 +39,19 @@ use hkdf::hmac::Hmac;
 ///
 /// # fn main() {
 ///     let ikm = b"02345678001234567890123456789012";
-///     let domain = b"my_app";
+///     let info = b"my_app";
 ///     let salt = b"some_salt";
 
-///     let my_keypair = hkdf_generate_from_ikm::<Sha3_256, BLS12381KeyPair>(ikm, salt, Some(domain)).unwrap();
-///     let native_sk = blst::min_sig::SecretKey::key_gen_v4_5(ikm, salt, domain).unwrap();
+///     let my_keypair = hkdf_generate_from_ikm::<Sha3_256, BLS12381KeyPair>(ikm, salt, info).unwrap();
+///     let native_sk = blst::min_sig::SecretKey::key_gen_v4_5(ikm, salt, info).unwrap();
 
 ///     assert_ne!(native_sk.to_bytes(), my_keypair.private().as_bytes());
 /// # }
 /// ```
-pub fn hkdf_generate_from_ikm<'a, H: OutputSizeUser, K: KeyPair>(
-    ikm: &[u8],             // IKM (32 bytes)
-    salt: &[u8],            // Optional salt
-    info: Option<&'a [u8]>, // Optional domain
+pub fn hkdf_generate_from_ikm<H, K>(
+    ikm: &[u8],  // IKM (32 bytes).
+    salt: &[u8], // Salt (can be empty).
```

Comment:
Ideally the salt should be an `Option`: checking the internals of the `hkdf` dep, and there is already a default salt if you add `None`. I guess this is preferable to comply with the underlying lib (which is already adheres to rfc5869), see https://github.com/RustCrypto/KDFs/blob/master/hkdf/src/lib.rs#L146

Reply:
Right, the library does that, which I find confusing since it also pads `salt` with zeros in case `salt` is shorter than the hash input length (as specified by HMAC). In other words, calling the external hkdf with `Some(&[])` or `None` results in the same output.

Reply:
Good point, I'd prefer if we had 

- a test to ensure the equivalence of both None and &[] just to ensure this assumption will stay true as the hkdf dep version evolves (in the future)
- add a comment line to explain what you mentioned above for future code readers.

Reply:
Done